debezium-server-kinesis/src/main/java/io/debezium/server/kinesis/KinesisChangeConsumer.java

```diff
@@ -101,6 +118,23 @@ public void handleBatch(List<ChangeEvent<Object, Object>> records, RecordCommitt
             throws InterruptedException {
         for (ChangeEvent<Object, Object> record : records) {
             LOGGER.trace("Received event '{}'", record);
+
+            int attempts = 0;
+            if (!recordSent(record)) {
```

Comment:
you need some while loop to retry sending the records when it fails, see e.g. https://github.com/debezium/debezium-server/blob/main/debezium-server-http/src/main/java/io/debezium/server/http/HttpChangeConsumer.java#L164